### PR TITLE
windows: Download dependencies, update build.bat

### PR DESCRIPTION
**Pull request by squalus**
* download gperf, bison, and ninja dependencies in getsrc step
* add support to extra_deps.ini for multiple artifacts to be extracted
  into the same path (for bison)
* update windows build script to use internal dependencies
* update windows build script to use vswhere.exe to locate Visual Studio
* fix random bugs in windows build script discovered in testing
* remove now-obsolete windows gperf/bison patch

### CODE REVIEW

**Comment by squalus** [comment]
Further automation for the Windows build.

- gperf, bison, and ninja no longer require ahead-of-time user setup
- build.bat script now works fine for me in testing

The bison zip files all extract to the same target directory, but extra_deps.ini section names still need to be distinct. To do this I introduced `:` as a delimiter to allow for an optional string distinguishing the otherwise identical entries.

Also I had SSL cert validation errors downloading from Sourceforge using official Python3.6 Windows binaries. I need to investigate this further, as it may be some local configuration problem. Linux/macOS works fine, and they don't need to download from Sourceforge anyway. Is it ok to disable SSL cert validation in the downloader, since we are checking hashes anyway?

**Comment by squalus** [comment]
On further thought, a better way to do the extra_deps.ini change might be to make the section heading an arbitrary string, e.g. "bison-bin", and introduce an "output_path" field

**Comment by Eloston** [comment]
I implemented your suggestion for output_path. Thanks :+1: 

> Also I had SSL cert validation errors downloading from Sourceforge using official Python3.6 Windows binaries. I need to investigate this further, as it may be some local configuration problem. Linux/macOS works fine, and they don't need to download from Sourceforge anyway. Is it ok to disable SSL cert validation in the downloader, since we are checking hashes anyway?

Yeah, I think that'll be fine. Just make sure to leave a comment in the code just in case anyone else gets confused by that decision.

File: buildkit/packaging/windows.py [hunk 1]
```diff
@@ -38,19 +38,14 @@ def generate_packaging(config_bundle, output_dir, build_output=DEFAULT_BUILD_OUT
...
-        build_output=build_output,
-        version_string=config_bundle.version.version_string
-    )
 
     ensure_empty_dir(output_dir) # Raises FileNotFoundError, FileExistsError
     (output_dir / 'scripts').mkdir()
 
     # Build and packaging scripts
-    _copy_from_resources('build.bat.in', output_dir)
+    _copy_from_resources('build.bat', output_dir)
```
[the remark this answers is not in the document]
Comment (squalus) [reply buildkit/packaging/windows.py]: There's no good reason for the change. I restored the build_output substitution in the build script.

**Comment by squalus** [comment]
Forgot the SSL certificate thing for now. I think it's just some kind of local misconfiguration on my machine. The downloads work fine on an Appveyor CI instance so I think it's just me. Besides, there's not a good way to disable certificate verification on Python3 urlretrieve. I'm ok for this to be merged.

**Pull request merged**